axelrod/tests/unit/test_pickling.py

```diff
@@ -238,19 +230,37 @@ def test_pickling_all_transformers_as_instance_called_on_a_class(self):
             self.assert_orignal_equals_pickled(player)
 
     def test_created_on_the_spot_multiple_transformers(self):
-        klass = st.FlipTransformer()(axl.Cooperator)
-        klass = st.DualTransformer()(klass)
-        player = st.FinalTransformer((C, D))(klass)()
+        player_class = st.FlipTransformer()(axl.Cooperator)
+        player_class = st.DualTransformer()(player_class)
+        player = st.FinalTransformer((C, D))(player_class)()
 
         self.assert_orignal_equals_pickled(player)
 
-    def test_dual_transformer_special_case(self):
-        player = TestDualTransformerIssues()
+    def test_dual_transformer_regression_test(self):
+        """DualTransformer has failed when there were multiple DualTransformers.
+        It has also failed when DualTransformer was not the outermost
+        transformer or when other transformers were between multiple
+        DualTransformers."""
+        player = InterspersedDualTransformersNamePrefixAbsent()
         self.assert_orignal_equals_pickled(player)
 
-        player = TestDualTransformerIssues2()
+        player = InterspersedDualTransformersNamePrefixPresent()
         self.assert_orignal_equals_pickled(player)
 
+        interspersed_dual_transformers = (
```

Comment:
This isn't terribly nice to look at. Perhaps we can make use of intermediate variables to improve readability?